src/test/java/pl/gemiusz/Case0022SetOrRefreshTokenSimulation.java

```diff
@@ -46,41 +48,59 @@ public class Case0022SetOrRefreshTokenSimulation extends Simulation {
                         return session;
                     });
 
+    ScenarioBuilder scnForSetFirstToken =
+            scenario("GeMi_SetFirstToken").exec(chainGenerateAndSetToken);
+
+    ScenarioBuilder scnForRefreshToken =
+            scenario("GeMi_SetOrRefreshToken")
+                    .exec(session -> session.set("tokenTime", 0))
+                    .asLongAs(session -> tokenRefreshContinue)
+                    .on(
+                            pause(Duration.ofSeconds(1))
+                                    .exec(session -> session.set("tokenTime", session.getInt("tokenTime") + 1))
+                                    .doIf(session -> session.getInt("tokenTime") % refreshTokenEverySeconds == 0)
```

Comment:
Matter of taste, I think.
I, personally, don't like to rely on an exact value when pauses are around.

What I have in mind:
 * in `chainGenerateAndSetToken`: set the current time (`java.time.Instant`?) when retrieving a valid token
 * here (in `scnForRefreshToken`): compare if the (time + duration is greater than `now()`)

WDYT?

Reply:
I was thinking about something similar but I don't be sure if I can use "getTime" so I used variable to count - need less resources.
I will prepare POC for that :)
Thank you! :)

Reply:
POC code commited -> https://github.com/gemiusz/gatling-examples-maven-java/commit/140d5515ddc054492c0df8ccb1885fda21379cdb 
Any suggestions are welcome :)